src/core/net/ip6.hpp

```diff
@@ -350,6 +350,26 @@ class Ip6 : public InstanceLocator, private NonCopyable
     void ResetBorderRoutingCounters(void) { ClearAllBytes(mBorderRoutingCounters); }
 #endif
 
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+
+    /**
+     * Enables or disables the filter that drops TMF UDP messages from untrusted origin.
+     *
+     * @param[in]  aEnabled  TRUE to enable filter, FALSE otherwise.
+     *
+     */
+    void SetTmfOriginFilterEnabled(bool aEnabled) { mTmfOriginFilterEnabled = aEnabled; }
+
+    /**
+     * Indicates whether the filter that drops TMF UDP messages from untrusted origin is enabled or not.
+     *
+     * @returns TRUE if the filter is enabled, FALSE otherwise.
+     *
+     */
+    bool IsTmfOriginFilterEnabled(void) { return mTmfOriginFilterEnabled; }
+
+#endif // OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
```

Comment:
Can remove ` // OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` in `#endif` if the number of lines to the corresponding `#if` is less than 20.